yeonjae/programmers-0/구슬을나누는경우의수.js

```diff
@@ -0,0 +1,16 @@
+function solution(balls, share) {
+  const numerator = factorial(balls);
+  const denominator = factorial(balls - share) * factorial(share);
+
+  function factorial(n) {
+    let i,
+      result = BigInt(1);
+
+    for (i = BigInt(2); i <= n; i++) {
+      result *= i;
+    }
+    return result;
+  }
+
+  return Number(numerator / denominator);
+}
```

Comment:
```suggestion
function solution(balls, share) {
  const numerator = factorial(balls);
  const denominator = factorial(balls - share) * factorial(share);

  function factorial(n) {
    let result = BigInt(1);

    for (let i = BigInt(2); i <= n; i++) {
      result *= i;
    }
    return result;
  }

  return Number(numerator / denominator);
}
```
- 풀이 잘봤습니다 👍
- for문을 돌리는 i를 굳이 외부에 선언하신 이유가 있나요?? 굳이 factorial 함수의 지역변수로 만드는 것보다 for문 내부에서만 사용할 수 있는 i를 선언하시는게 좋습니다 혹시라도 다른 이유가 있다면 말씀해주세여 ~!

Reply:
팩토리얼 함수를 만들때 초기값으로 bigInt로 설정한다는 생각으로 세팅해줬던건데, i를 굳이 밖으로 빼서 선언할필요는 없었던것같습니다! 
피드백 감사합니다~!